internal/controller/dscinitialization/dscinitialization_controller.go

```diff
@@ -417,16 +441,20 @@ func (r *DSCInitializationReconciler) newMonitoringCR(ctx context.Context, dsci
 		},
 	}
 
+	// if metrics is not removed from DSCI totally, otherwise it is default as nil in defaultMonitoring
 	if dsci.Spec.Monitoring.Metrics != nil {
-		defaultMonitoring.Spec.Metrics = dsci.Spec.Monitoring.Metrics
+		// when metrics has values set in resoures or storage. skip replicas since it cannot be 0 from CEL validation
+		if dsci.Spec.Monitoring.Metrics.Storage != nil || dsci.Spec.Monitoring.Metrics.Resources != nil {
+			defaultMonitoring.Spec.Metrics = dsci.Spec.Monitoring.Metrics
+		} else { // if metrics is set to metrics:{} to avoid  invalid value "null" to patch existing Monitoring CR
+			defaultMonitoring.Spec.Metrics = nil // explictliy set to nil, same as not set but for better readability
```

Comment:
Replicas can be zero if storage and resources are nil.
However, here:
https://github.com/opendatahub-io/opendatahub-operator/blob/bcd8c13a0110ea3d649fc2e3eb97da3902a4c7b7/internal/controller/services/monitoring/monitoring_controller_actions.go#L96
we deploy MonitoringStack if storage and resources is nil but replicas is not zero. But in in this PR we just check resources and storage. We should probably remove the replica check from there as well.